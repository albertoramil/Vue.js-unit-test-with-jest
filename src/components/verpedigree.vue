
<template>
  <div>
    <i>{{hint}}</i>

    <v-text-field type="input" data-test="zipCodeText" v-model="inputText"></v-text-field>

<!--
    <v-text-field
      v-model="email"
      :rules="emailRules"
      label=" label=LABEL"
      required
    ></v-text-field>-->


    <v-text-field
      type="input"
      :rules="[v => !!v || '']"
      maxlength="25"
      hint="Tama침o m치ximo"
      data-test="entradaperro"
      v-model="perro"
    ></v-text-field>

    <v-btn v-on:click="save" data-test="saveButton" color="primary">Enter</v-btn>
    <v-btn
      color="success"
      data-test="handleClickButton"
      class="mr-4"
      @click="handleClick"
    >Ver Pedigree</v-btn>

    <br />
  </div>
</template>

<script>
export default {
  props: {
    hint: {
      type: String
    }
  },
  data() {
    return {
      inputText: "",
      perro: "",
      url:"",
      count: 0,

    };
  },
  methods: {
    save() {
      this.$emit("save", this.inputText);
      this.inputText = "";
    },
    increment () {
      this.count++
    },
    handleClick() {
      var valor = this.perro;
      console.log(valor);


      switch (valor) {
        case "cora": 
                    this.count++

          location.href ="https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=11475&fn=pedigree&rq=CORA";
                this.url="https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=11475&fn=pedigree&rq=CORA"

          break;

        case "imba":
          location.href ="https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=32657";
          url=location.href
          break;

        case "lula":
          location.href ="https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=48852";
          url=location.href
          break;

        case "lupin":
          location.href ="https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=48853&fn=pedigree&rq=LUPIN+DE+LA+BARRERA";
          url=location.href
          break;

        default:
          break;
      }
    }
  }
};
</script>

<style>
.inputBox {
  height: 30px;
  width: 200px;
  border: 5px;
}
</style>









<!--
<template>
  <v-row align="center">
    <v-form
      ref="form"
      v-model="valid">

      <v-text-field
        v-model="perro"
        :counter="10"
        :rules="nameRules"
        label="Perro"
        required
      ></v-text-field>
      <v-btn
        color="success"
        class="mr-4"
        @click="handleClick"
      >
        Ver Pedigree
      </v-btn>     
    </v-form>
  </v-row>
</template>


<template>
    <v-container>
        <v-text-field
              type="input"

        data-test="entradaPerro"
          v-model="perro"
          :rules="[v => !!v || '']"
          maxlength="25"
          hint="Tama침o m치ximo"
          label="Introduce el nombre del perro"
        ></v-text-field>

        <v-btn  color="success" data-test="handleClickButton" class="mr-4" @click="handleClick">Ver Pedigree</v-btn>
    </v-container>
</template>




<script>
export default {
  data: () => {
   return {
      perro: "",
          habilitar: true,

      
    }
  },

  methods: {
    handleClick: function(event) {
      var valor = this.perro;
      console.log(valor);
      switch (valor) {
        case "cora":
          location.href =
            "https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=11475&fn=pedigree&rq=CORA";
          break;

        case "imba":
          location.href =
            "https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=32657";
          break;

        case "lula":
          location.href =
            "https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=48852";
          break;

        case "lupin":
          location.href =
            "https://pedigree.setter-anglais.fr/genealogie/arbre.php?id=48853&fn=pedigree&rq=LUPIN+DE+LA+BARRERA";
          break;

        default:
          break;
      }
    },save(){
      this.$emit('save', this.perro)
      this.perro = "";
    }
  }
};
</script>
<style scoped>
.container {
  max-width: 1500px;
  margin: auto;
  padding: 15px 30px;
  /*  background-color: rgb(136,126,0);*/
  font-size: 180%;
}
</style>
-->
