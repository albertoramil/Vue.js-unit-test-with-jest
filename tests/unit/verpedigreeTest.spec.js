//tests/ ZipCodeInput.spec.js

import { createLocalVue, mount } from '@vue/test-utils'
import verpedigree from '@/components/verpedigree.vue'

describe('ZipCodeInput', () => {

   const wrapper = mount(verpedigree    )
  

  it('Find input- type text ', ()=>{
    expect(wrapper.contains('[        data-test="entradaPerro"  ]')).toBe(true)
  })
  it('has a button', ()=>{
    expect(wrapper.contains('[data-test="handleClickButton"]')).toBe(true)

  })
})

