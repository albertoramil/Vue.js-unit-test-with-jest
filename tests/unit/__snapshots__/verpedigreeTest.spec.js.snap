// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard.vue should have a custom title and match snapshot 1`] = `
<v-card>
  <v-card-title><span>Foobar</span>
    <v-spacer></v-spacer>
    <v-btn>
      Action
    </v-btn>
  </v-card-title>
  <v-card-text></v-card-text>
</v-card>
`;

exports[`verpedigree captura de los elementos html 1`] = `
<v-form>
  <v-container>
    <v-form>
      <v-text-field rules="function (v) {
        /* istanbul ignore next */cov_hlhqjoykk.f[5]++;
        cov_hlhqjoykk.s[26]++;
        return (/* istanbul ignore next */(cov_hlhqjoykk.b[4][0]++, !!v) || /* istanbul ignore next */(cov_hlhqjoykk.b[4][1]++, '')
        );
      }" maxlength="25" hint="Tama침o m치ximo" label="Introduce el nombre del perro"></v-text-field>
      <v-btn color="success" class="mr-4">
        Ver Pedigree
      </v-btn>
    </v-form>
  </v-container>
</v-form>
`;

exports[`verpedigree existe el boton y le puedo dar 1`] = `
<v-form>
  <v-container>
    <v-form>
      <v-text-field rules="function (v) {
        /* istanbul ignore next */cov_vfbk8k1kf.f[5]++;
        cov_vfbk8k1kf.s[26]++;
        return (/* istanbul ignore next */(cov_vfbk8k1kf.b[4][0]++, !!v) || /* istanbul ignore next */(cov_vfbk8k1kf.b[4][1]++, '')
        );
      }" maxlength="25" hint="Tama침o m치ximo" label="Introduce el nombre del perro"></v-text-field>
      <v-btn color="success" class="mr-4">
        Ver Pedigree
      </v-btn>
    </v-form>
  </v-container>
</v-form>
`;
